src/main/java/org/Olivia/GuiEventDispatcher.java

```diff
@@ -0,0 +1,177 @@
+            //return 400;
+        }
+        if (args[0].equals("mark")) {
+            int status = this.table.markAsDone(Integer.parseInt(args[1]));
+            if (status == 200 || status == 208) {
+                disk.syncToFile(this.table);
+                return "Jawohl, I have marked the following event as completed:\n     " +
+                        this.table.getEntry(Integer.parseInt(args[1])) + "\n";
+            }
+
+        }
+        if (args[0].equals("unmark")) {
+            int status = this.table.markAsUndone(Integer.parseInt(args[1]));
+            if (status == 200 || status == 208) {
+                disk.syncToFile(this.table);
+                return "Jawohl, I have marked the following event as incomplete:\n     " +
+                        this.table.getEntry(Integer.parseInt(args[1])) + "\n";
+            }
+        }
+        throw new RuntimeException("Sorry, there seems to be some difficulties processing your command, could you check the syntax and try again later?");
+    }
+
+    private String delete(String input) throws Exception {
+        String[] args = input.toLowerCase().split(" ");
+        if (args.length != 2) {
+            throw new InvalidParameterException("Sorry, which entry do you want me to delete?");
+            //return 400;
+        }
+        disk.syncToFile(this.table);
+        return "Jawohl, I have removed the following entry from your calendar:\n     " +
+                this.table.deleteEntry(Integer.parseInt(args[1])) + "\n";
+    }
+
+    private String addEntryToCalendar(String input) throws Exception {
+        String[] args = input.toLowerCase().split(" ");
+        if (args[0].equals("todo") && args.length >= 2) {
+            CalendarEntryTodo entry = new CalendarEntryTodo(input.substring(5));
+            int status = this.table.addEntry(entry);
+            if (status == 200) {
+                disk.syncToFile(this.table);
+                return "Verstehe, added: " + entry.toString() + "\n";
+            }
+
+        }
+        else if (args[0].equals("deadline") && args.length >= 4) {
+            if (input.indexOf("/by") == -1) {
+                throw new InvalidParameterException("Sorry, what is the exact time of the deadline?\nCheck the help message for information on command syntax");
+                //return 500;
+            }
+            input = input.substring(9);
+            String time = input.substring(input.indexOf("/by") + 4);
+            String title = input.substring(0, input.indexOf("/by") - 1);
+            CalendarEntryDeadline entry = new CalendarEntryDeadline(title, time);
+            int status = this.table.addEntry(entry);
+            if (status == 200) {
+                disk.syncToFile(this.table);
+                return "Verstehe, added: " + entry.toString() + "\n";
+            }
+
+        }
+        else if (args[0].equals("event") && args.length >= 4) {
```

Comment:
I think it should be as following:
```suggestion
        } else if (args[0].equals("event") && args.length >= 4) {
```